plugins/notifications-backend/src/service/types.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { TokenManager } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
+import { Config } from '@backstage/config';
+import { IdentityApi } from '@backstage/plugin-auth-node';
+import { PermissionEvaluator } from '@backstage/plugin-permission-common';
+
+import { Logger } from 'winston';
+
+export interface RouterOptions {
+  logger: Logger;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
+  identity: IdentityApi;
+  permissions: PermissionEvaluator;
+  tokenManager: TokenManager;
+
+  // Workaround - see auth.ts
+  externalCallerSecret?: string;
+}
+
+export type NotificationsFilterRequest = {
+  /**
+   * Filter notifications whose either title or message contains the provided string.
+   */
+  containsText?: string;
+
+  /**
+   * Only notifications created after this timestamp will be included.
+   */
+  createdAfter?: string;
+
+  /**
+   * See MessageScopes
+   * Default: DefaultMessageScope
+   */
+  messageScope?: string;
+
+  /**
+   * 'false' for user's unread messages, 'true' for read ones.
+   * If undefined, then both marks.
+   */
+  read?: boolean;
+};
+
+/**
+ * How the result set is sorted.
+ */
+export type NotificationsSortingRequest = {
+  orderBy?: string;
+  OrderByDirec?: string;
```

Comment:
I think these are usually `orderBy` and `order`. The naming convention of this is not very clear.

Reply:
Five projects, five naming conventions for this. I am ok with aligning to whatever here.
Just the `order` is not saying explicitly that it is a direction.

Reply:
Maybe the catalog ordering can be good inspiration.

https://backstage.io/docs/features/software-catalog/software-catalog-api#ordering

They suffixed it `Field` to make room for completely different sorts (hah!) of sorting down the line; that may not be needed here. So maybe

`order=<what>,<dir>`?

Reply:
it is a shame that the name of a variable causes such a long discussion. I did not come up with this name but when I saw it for the first time it was obvious what it is used for.